lib/index.js

```diff
@@ -2,35 +2,40 @@
 
 const Boom = require('boom');
 const Hoek = require('hoek');
+const Joi = require('joi');
 
 // Declare Internals
 
 const internals = {};
 
-exports.register = (server, options, next) => {
-
-    server.auth.scheme('bearer-access-token', internals.implementation);
-    next();
+internals.defaults = {
+    accessTokenName: 'access_token',
+    allowQueryToken: true,
+    allowCookieToken: false,
+    allowMultipleHeaders: false,
+    allowChaining: false,
+    tokenType: 'Bearer'
 };
 
-exports.register.attributes = {
-    pkg: require('../package.json')
-};
+internals.schema = Joi.object().keys({
+    validateFunc: Joi.func().required(),
+    accessTokenName: Joi.string().required(),
+    allowQueryToken: Joi.boolean(),
+    allowCookieToken: Joi.boolean(),
+    allowMultipleHeaders: Joi.boolean(),
+    allowChaining: Joi.boolean(),
+    tokenType: Joi.string().required()
+});
 
 internals.implementation = (server, options) => {
 
     Hoek.assert(options, 'Missing bearer auth strategy options');
-    Hoek.assert(typeof options.validateFunc === 'function', 'options.validateFunc must be a valid function in bearer scheme');
 
-    options.accessTokenName = options.accessTokenName || 'access_token';
-    options.allowQueryToken = options.allowQueryToken === false ? false : true;
-    options.allowCookieToken = options.allowCookieToken === true ? true : false;
-    options.allowMultipleHeaders = options.allowMultipleHeaders === true ? true : false;
-    options.tokenType = options.tokenType || 'Bearer';
+    const settings = Hoek.applyToDefaults(internals.defaults, options);
```

Comment:
Here there are a couple of examples: https://github.com/hapijs/heavy/blob/master/lib/index.js#L44, https://github.com/hapijs/catbox/blob/master/lib/client.js#L26 https://github.com/hapijs/hapi/blob/master/lib/server.js#L36